src/utils/resize.observer.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react'
+
+class ResizeObserver extends Component {
+  container = React.createRef()
+
+  state = {
+    width: null,
+  }
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize)
+    this.setState({
+      width: this.container.current.getBoundingClientRect().width,
+    })
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize)
+  }
+
+  handleResize = () => {
```

Comment:
do we need to debounce?? doesn't looks like that it will trigger changes very often